### PR TITLE
add MTU settings

### PR DESCRIPTION
**Pull request by crvv**
According to the document, MTU should be 1460.

https://cloud.google.com/compute/docs/troubleshooting/general-tips

### CODE REVIEW

File: build-arch-gce [hunk 1]
```diff
@@ -139,6 +139,11 @@ arch-chroot -- "$mount_dir" /bin/bash -s -- "$loop_dev" <<-'EOS'
 		google-accounts-daemon google-clock-skew-daemon google-instance-setup \
 		google-network-daemon google-shutdown-scripts google-startup-scripts
 
+	echo "-- setting mtu"
+	cat <<-'EOF' > /etc/udev/rules.d/10-network.rules
+		ACTION=="add", SUBSYSTEM=="net", KERNEL=="eth*", ATTR{mtu}="1460"
```
Comment (toastwaffle) [inline build-arch-gce]: On the GCE instance I have, the network interface is ens4, not ethX. I think "en*" should be sufficient here per https://github.com/systemd/systemd/blob/master/src/udev/udev-builtin-net_id.c#L13

```suggestion
		ACTION=="add", SUBSYSTEM=="net", KERNEL=="en*", ATTR{mtu}="1460"
```

File: build-arch-gce [hunk 2]
```diff
@@ -139,6 +139,11 @@ arch-chroot -- "$mount_dir" /bin/bash -s -- "$loop_dev" <<-'EOS'
 		google-accounts-daemon google-clock-skew-daemon google-instance-setup \
 		google-network-daemon google-shutdown-scripts google-startup-scripts
 
+	echo "-- setting mtu"
```
Comment (toastwaffle) [inline build-arch-gce]: nit

```suggestion
	echo "-- setting MTU"
```

Comment (crvv) [reply build-arch-gce]: Done

File: build-arch-gce (hunk 1 above)
Comment (crvv) [reply build-arch-gce]: It is "ens4" on my instance, too.
And "eth*" and "en*" both work.
I have updated this to "en*".

**Comment by toastwaffle** [comment]
Thank you for your contribution!

**Comment by lcastelli** [comment]
This should be set automatically via DHCP. Did you see this being set incorrectly before your CL?

**Comment by crvv** [comment]
Yes, it was 1500.

And the document says
> Operating system images provided by Compute Engine are configured with the appropriate MTU, so no action is required if you're using one of those images. For custom images, set the MTU to 1460 for custom Linux images and Windows Server images to avoid the increased latency and packet overhead caused by fragmentation.

Maybe Google should add this setting to the DHCP server.

**Comment by lcastelli** [comment]
It should already be set by DHCP, and indeed it works for me. I'm not sure why it doesn't for you.

Can you `cat /var/lib/dhclient/dhclient.leases`, and see if that contains "option interface-mtu 1460;"?

**Comment by crvv** [comment]
It didn't work because I used dhcpcd.

I tested this today and it indeed works if I use dhclient.
And the right MTU is included in the DHCP response.

If I run
```
sudo systemctl enable dhcpcd
sudo systemctl disable dhclient@ens4
sudo reboot
```
the MTU of ens4 will be 1500.

The following request is sent by dhcpcd.
It requested MTU and got 1460, but I don't know why it doesn't set the interface MTU to the right value.

```
$ sudo tcpdump -i ens4 -n port 67 and port 68 -vvv
tcpdump: listening on ens4, link-type EN10MB (Ethernet), capture size 262144 bytes
10:11:16.524598 IP (tos 0x0, ttl 64, id 7447, offset 0, flags [none], proto UDP (17), length 395)
    0.0.0.0.68 > 255.255.255.255.67: [udp sum ok] BOOTP/DHCP, Request from 42:01:0a:8c:00:0b, length 367, xid 0x7ed1ec46, Flags [none] (0x0000)
          Client-Ethernet-Address 42:01:0a:8c:00:0b
          Vendor-rfc1048 Extensions
            Magic Cookie 0x63825363
            DHCP-Message Option 53, length 1: Request
            Client-ID Option 61, length 23: hardware-type 255, 0a:8c:00:0b:00:04:22:71:61:84:58:5a:17:72:18:f8:79:b2:d3:86:e4:0a
            Requested-IP Option 50, length 4: 10.140.0.11
            MSZ Option 57, length 2: 1472
            Vendor-Class Option 60, length 57: "dhcpcd-7.0.8:Linux-5.0.4-arch1-1-ARCH:x86_64:GenuineIntel"
            Hostname Option 12, length 8: "template"
            T145 Option 145, length 1: 1
            Parameter-Request Option 55, length 14:
              Subnet-Mask, Classless-Static-Route, Static-Route, Default-Gateway
              Domain-Name-Server, Hostname, Domain-Name, MTU
              BR, Lease-Time, Server-ID, RN
              RB, Option 119
            END Option 255, length 0
10:11:16.525130 IP (tos 0x0, ttl 1, id 0, offset 0, flags [none], proto UDP (17), length 454)
    169.254.169.254.67 > 10.140.0.11.68: [udp sum ok] BOOTP/DHCP, Reply, length 426, xid 0x7ed1ec46, Flags [none] (0x0000)
          Your-IP 10.140.0.11
          Server-IP 10.140.0.1
          Gateway-IP 10.140.0.1
          Client-Ethernet-Address 42:01:0a:8c:00:0b
          Vendor-rfc1048 Extensions
            Magic Cookie 0x63825363
            DHCP-Message Option 53, length 1: ACK
            Server-ID Option 54, length 4: 169.254.169.254
            Domain-Name-Server Option 6, length 4: 169.254.169.254
            Lease-Time Option 51, length 4: 86400
            Domain-Name Option 15, length 32: ""
            T119 Option 119, length 51: 1.99.21.102.101.105.115.116.121.45.101.116.99.104.105.110.103.45.49.57.48.53.48.54.8.105.110.116.101.114.110.97.108.0.6.103.111.111.103.108.101.8.105.110.116.101.114.110.97.108.0
            Subnet-Mask Option 1, length 4: 255.255.255.255
            Default-Gateway Option 3, length 4: 10.140.0.1
            Classless-Static-Route Option 121, length 14: (10.140.0.1/32:0.0.0.0),(default:10.140.0.1)
            MTU Option 26, length 2: 1460
            Hostname Option 12, length 37: ""
            NTP Option 42, length 4: 169.254.169.254
            END Option 255, length 0
```

Anyway, I think the change should be reverted. Sorry for disturbing.

**Comment by crvv** [comment]
I finally understand what happens here.
`dhcpcd` doesn't set the MTU of interfaces.
It only set MTU of routes.

I can use `ip route` to get the MTU set by `dhcpcd`

**Comment by lcastelli** [comment]
That explains, thanks for checking. BTW the Linux Guest Environment package also has [assumptions](https://github.com/GoogleCloudPlatform/compute-image-packages/blob/6767372dfedcb8ee52ed849732678e86e87657ae/packages/python-google-compute-engine/google_compute_engine/networking/network_setup/network_setup.py#L31) that network is configured through dhclient, so I'd recommend using that if you can.

**Pull request merged**